halotools/mock_observables/counts_in_cells/engines/counts_in_cylinders_engine.pyx

```diff
@@ -238,7 +268,11 @@ def counts_in_cylinders_engine(
                                     dz_sq = dz*dz
 
                                     if (dxy_sq < rp_max_squaredtmp) & (dz_sq < pi_max_squaredtmp):
-                                        counts[ifirst1+i] += 1
+
+                                        i_original = idx_sorted1[ifirst1+i]
+                                        j_original = idx_sorted2[ifirst2+j]
+                                        if cond.func(i_original, j_original):
+                                            counts[ifirst1+i] += 1
 
                                         if c_return_indexes:
```

Comment:
Does this line also need to be modified in order to get the condition function to work properly with the return_indexes feature?